sim/target-agnostic.mk

```diff
@@ -51,8 +53,16 @@ firesim_root_sbt_project := {file:$(firesim_base_dir)}firesim
 # extracted used to generate new runtime configurations.
 fame_annos := $(GENERATED_DIR)/post-bridge-extraction.json
 
-$(VERILOG) $(HEADER) $(fame_annos): $(FIRRTL_FILE) $(ANNO_FILE) $(SCALA_BUILDTOOL_DEPS)
-	$(call run_scala_main,$(firesim_sbt_project),midas.stage.GoldenGateMain,\
+$(GENERATED_DIR):
+	mkdir -p $@
+
+# empty recipe to help make understand multiple targets come from single recipe invocation
+# without using the new (4.3) '&:' grouped targets see https://stackoverflow.com/a/41710495
+$(VERILOG) $(HEADER) $(fame_annos): $(VERILOG).intermediate
+
+.SECONDARY: $(VERILOG).intermediate
+$(VERILOG).intermediate: $(FIRRTL_FILE) $(ANNO_FILE) $(RUN_MAIN_DEPS)
```

Comment:
Here's another problem.  I renamed `RUN_MAIN_DEPS` to `PRE_ELABORATION_DEPS` and forgot to change it here.